### PR TITLE
Cleanup for NodeFeature API being GA

### PR DESCRIPTION
**Pull request by marquiz**
Drop references to the gRPC API and don't suggest that NodeFeatureAPI could be disabled.

Also update the developer guide for instructions running nfd components outside the cluster.

### CODE REVIEW

File: docs/developer-guide/index.md [hunk 1]
```diff
@@ -173,58 +173,53 @@ e2e-tests:
...
 
 ### NFD-Topology-Updater
 
-To run nfd-topology-updater as a "stand-alone" container
-you need to run it in with the `-no-publish` flag to disable communication to
-the Kubernetes apiserver.
+For development and debugging it is possible to run nfd-topology-updater as a
+stand-alone binary outside the cluster. However, it requires access to the
+kubelet's local pod-resources socket and the kubelet http api so in practice it
+needs to be run on a host acrting as a Kubernetes node and thus running
```
[the remark this answers is not in the document]
Comment (marquiz) [reply docs/developer-guide/index.md]: Sharp eyes, thanks @fmuyassarov. Fixed

**Comment by marquiz** [comment]
Updated. @fmuyassarov PTAL

File: docs/developer-guide/index.md [hunk 2]
```diff
@@ -173,58 +173,53 @@ e2e-tests:
 
 ### NFD-Master
 
-When running as a standalone container labeling is expected to fail because
-Kubernetes API is not available. Thus, it is recommended to use `-no-publish`.
+For development and debugging it is possible to run nfd-master as a stand-alone
+binary outside the cluster. The `-no-publish` flag can be used to prevent
+nfd-master making changes to the nodes. If `-no-publish` is not set, nfd-master
+also requires the `NODE_NAME` environment variable to be set for cleaning up
+stale annotations.
```
Comment (fmuyassarov) [inline docs/developer-guide/index.md:180]: quick question, `KUBERNETES_NAMESPACE` isn't needed for the master?

Comment (marquiz) [reply docs/developer-guide/index.md:180]: Good question. And correct, it is not needed for nfd-master (the namespace env is required for those daemons that create CRs). The nfd-master watches CRs in all namespaces

**Pull request merged**